cachito/web/content_manifest.py

```diff
@@ -97,13 +97,23 @@ def process_gomod(self, package, dependency, type="icm"):
 
             # if we're processing a local dependency
             if dependency.version and dependency.version.startswith("."):
+                match_type = None
                 dep_normpath = os.path.normpath(os.path.join(package.name, dependency.version))
-                parent_module_name = gomod.match_parent_module(
+                if parent_module_name := gomod.match_parent_module(
+                    dependency.name, self._gomod_data.keys()
+                ):
+                    match_type = "name"
+                elif parent_module_name := gomod.match_parent_module(
                     dep_normpath, self._gomod_data.keys()
-                )
+                ):
+                    match_type = "version"
                 # if the dependency is not in this repo, use a relative path from the root module
                 # otherwise just use the parent purl directly
-                if dependency.name not in self._gomod_data:
+                if dependency.name not in self._gomod_data and match_type == "name":
```

Comment:
And we can replace the `if dependency.name not in self._gomod_data` logic with

```python
if relpath_from_parent_module_to_dep == Path():
    relpath_from_parent_module_to_dep = None
```
That should mean the same thing - if the dependency name is in _gomod_data, then it will match itself => the relpath will be empty